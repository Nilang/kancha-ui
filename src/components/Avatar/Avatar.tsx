import * as React from 'react'
import { Image, ImageSourcePropType } from 'react-native'

import Text from '../Text/Text'
import Container from '../Container/Container'
import { withTheme } from '../../theming'

interface AvatarProps {
  /**
   * Title for avatar. Gets shortened to first 2 letters. eg.'Sa' Useful for usernames.
   */
  title?: string

  /**
   * Avatar size. default 32
   */
  size?: number

  /**
   * Rounded or squared.
   */
  type?: 'square' | 'rounded'

  /**
   * Standard image source type prop
   */
  source?: ImageSourcePropType

  /**
   * Provide custom background color. Only visible in text avatars
   */
  backgroundColor?: string

  /**
   * Theme properties passed in by HOC. Can ignore.
   */
  theme: any
}

const Avatar: React.FC<AvatarProps> = props => {
  const generateTitle = (str: string) => {
    return str.charAt(0).toUpperCase() + str.charAt(1).toLowerCase()
  }

  const size = props.size ? props.size : 32
  const type = props.type ? props.type : 'rounded'
  const title = props.title ? generateTitle(props.title) : '?'

  const avatarTypeStyle = {
    height: size,
    width: size,
    backgroundColor: props.backgroundColor ? props.backgroundColor : props.theme.colors.primary.brand,
    ...(type === 'square' ? { borderRadius: 5 } : { borderRadius: size / 2 }),
  }

  const imageAvatar = props.source ? <Image source={props.source} style={avatarTypeStyle} /> : null
  const textAvatar = (
    <Container viewStyle={avatarTypeStyle} alignItems={'center'} justifyContent={'center'}>
      <Text bold={true} textColor={'#ffffff'} textStyle={{ fontSize: size * 0.5 }}>
        {title}
      </Text>
    </Container>
  )

  return props.source ? imageAvatar : textAvatar
}

export default withTheme(Avatar)
