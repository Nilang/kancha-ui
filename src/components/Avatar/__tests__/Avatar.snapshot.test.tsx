import * as React from 'react'
import { render } from 'react-native-testing-library'
import Avatar from '../Avatar'

describe('Component(snapshots): Avatar', () => {
  it('should render with no props', () => {
    const tree = render(<Avatar />)

    expect(tree).toMatchSnapshot()
  })

  it('should render with source props', () => {
    const tree = render(<Avatar source={{ uri: 'somehttp-image-uri' }} />)

    expect(tree).toMatchSnapshot()
  })

  it('should render with props', () => {
    const tree = render(<Avatar source={{ uri: 'somehttp-image-uri' }} size={32} type={'square'} />)

    expect(tree).toMatchSnapshot()
  })

  it('should render with title and background props', () => {
    const tree = render(<Avatar size={32} type={'square'} title={'Satoshi'} backgroundColor={'#000000'} />)

    expect(tree).toMatchSnapshot()
  })
})
