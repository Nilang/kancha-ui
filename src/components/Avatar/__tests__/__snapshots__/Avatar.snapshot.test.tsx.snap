// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(snapshots): Avatar should render with no props 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0078ff",
      "borderRadius": 16,
      "flex": undefined,
      "flexDirection": undefined,
      "height": 32,
      "justifyContent": "center",
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "width": 32,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    ?
  </Text>
</View>
`;

exports[`Component(snapshots): Avatar should render with props 1`] = `
<Image
  source={
    Object {
      "uri": "somehttp-image-uri",
    }
  }
  style={
    Object {
      "backgroundColor": "#0078ff",
      "borderRadius": 5,
      "height": 32,
      "width": 32,
    }
  }
/>
`;

exports[`Component(snapshots): Avatar should render with source props 1`] = `
<Image
  source={
    Object {
      "uri": "somehttp-image-uri",
    }
  }
  style={
    Object {
      "backgroundColor": "#0078ff",
      "borderRadius": 16,
      "height": 32,
      "width": 32,
    }
  }
/>
`;

exports[`Component(snapshots): Avatar should render with title and background props 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "borderRadius": 5,
      "flex": undefined,
      "flexDirection": undefined,
      "height": 32,
      "justifyContent": "center",
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "width": 32,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Sa
  </Text>
</View>
`;
